var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -0,0 +1,133 @@
+
+from spack import *
+import os
+
+
+class Timemory(CMakePackage):
+    """Timing + Memory + Hardware Counter Utilities for C / C++ / CUDA / Python"""
+
+    homepage = "https://timemory.readthedocs.io/en/latest/"
+    git = "https://github.com/NERSC/timemory.git"
+
+    version('master', branch='master', submodules=True)
+
+    variant('python', default=True, description="Enable Python support")
+    variant('mpi', default=True, description="Enable MPI support")
+    variant('papi', default=True, description="Enable PAPI support")
+    variant('cuda', default=True, description="Enable CUDA support")
+    variant('caliper', default=True, description="Enable Caliper support")
+    variant('gperftools', default=True, description="Enable gperftools support")
+    variant('cupti', default=True, description="Enable CUPTI support")
+
+    extends('python', when="+python")
+    depends_on('python@3:', when="+python")
+    depends_on('py-pip', when='+python')
+    depends_on('mpi', when='+mpi')
+    depends_on('papi', when='+papi')
+    depends_on('cuda', when='+cuda')
+    depends_on('gperftools', when='+gperftools')
+    depends_on('caliper', when='+caliper')
+    depends_on('gperftools', when='+gperftools')
+
+    parallel = True
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        """Set PYTHONPATH to include the site-packages directory for the
+        extension and any other python extensions it depends on."""
+
+        if '+python' in self.spec:
+            # If we set PYTHONHOME, we must also ensure that the corresponding
+            # python is found in the build environment. This to prevent cases
+            # where a system provided python is run against the standard libraries
+            # of a Spack built python. See issue #7128
+            spack_env.set('PYTHONHOME', self.home)
+
+            path = os.path.dirname(self.command.path)
+            if not is_system_path(path):
+                spack_env.prepend_path('PATH', path)
+
+            python_paths = []
+            for d in dependent_spec.traverse(
+                    deptype=('build', 'run', 'test')):
+                if d.package.extends(self.spec):
+                    python_paths.append(join_path(d.prefix,
+                                                  self.site_packages_dir))
+
+            pythonpath = ':'.join(python_paths)
+            spack_env.set('PYTHONPATH', pythonpath)
+
+            # For run time environment set only the path for
+            # dependent_spec and prepend it to PYTHONPATH
+            if dependent_spec.package.extends(self.spec):
+                run_env.prepend_path('PYTHONPATH', join_path(
+                    dependent_spec.prefix, self.site_packages_dir))
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Use spack install of Caliper instead of internal build
+        args = [
+            "-DTIMEMORY_BUILD_CALIPER=OFF",
```

Comment:
No. There is a git-submodule for Caliper but since an install is available via Spack, setting `TIMEMORY_BUILD_CALIPER=OFF` tells CMake to skip the `git submodule update --init external/caliper` but `TIMEMORY_USE_CALIPER=ON` means just do a `find_package(Caliper)`.